src/components/listitems/ListItemSwitch.tsx

```diff
@@ -10,14 +15,25 @@ import { IOIcons, Icon } from "../icons";
 import { HSpacer, VSpacer } from "../spacer";
 import { H6, LabelSmall, LabelLink } from "../typography";
 import { NativeSwitch } from "../switch/NativeSwitch";
+import { Badge } from "../badge";
 
 type Props = {
   label: string;
   onSwitchValueChange?: (newValue: boolean) => void;
   description?: string;
   icon?: IOIcons;
   action?: SwitchAction;
-};
+  isLoading?: boolean;
+} & (
+  | {
+      badgeText: string;
+      badgeVariant: Pick<Badge, "variant">["variant"];
```

Comment:
What about using an `object` instead of two different props, like the `action` above? So the code below could be slightly clearer (`badge` instead of `badgeText`):

```jsx
{!isLoading && !badge && (
```

Reply:
I would suggest to keep the strong handling for the typedef.

Something like this could be accurate:
```typescript
{
  ...rest
} & (
  | {
      badge: {
        badgeText: string;
        badgeVariant: Pick<Badge, "variant">["variant"];
      };
    }
  | {
      badge?: never;
    }
)
```

Reply:
What do you think about the implementation addressed into https://github.com/pagopa/io-app-design-system/pull/79/commits/91c4bddee31b2f61f06d7d8a2a378f1130c601dc?

Since the `Badge` type have only 2 required props (text and variant) and an optional property that is the testID (that could be useful in some test cases), in this case there is no more to handle the conditional typedef. What do you think?